components/sup/src/main.rs

```diff
@@ -160,10 +161,11 @@ fn sub_bash() -> Result<()> { command::shell::bash() }
 /// * `MemberList::initial_members` (write)
 /// * `MemberList::entries` (write)
 /// * `GatewayState::inner` (write)
-fn sub_run_rsr_imlw_mlw_gsw(m: &ArgMatches,
-                            launcher: LauncherCli,
-                            feature_flags: FeatureFlag)
-                            -> Result<()> {
+/// * `RumorHeat::inner` (write)
+fn sub_run_rsr_imlw_mlw_gsw_rhw(m: &ArgMatches,
```

Comment:
Also wondering if this function really needs annotations, given that it's what actually starts the supervisor (thus actually creating all the data structures in question). Everything that happens in the Supervisor ultimately happens _inside_ this function call; once it returns, the whole process ends, basically.

(same question applies up and down this call chain, too)

Reply:
Given all of the existing annotations, I'm going with yes.  If we want to change the naming of this, ok, but for the sake of this particular PR, I think following the existing pattern is appropriate.

Reply:
I would say the only place that annotations _shouldn't_ propagate back is when a new thread is started. That's why all the `run_loop() -> !` functions don't have annotations.